apps/graphql/src/apps/location/dataloaders/LocationsByID.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import stringify from 'json-stable-stringify';
+import { OptimisticDataloader } from '@kiwicom/graphql-utils';
+
+import fetch from '../../../services/fetch/tequilaFetch';
+import { type Location } from '../Location';
+import sanitizeLocations from '../helpers/sanitizeLocations';
+
+export type LocationInputID = {|
+  +code: string,
+|};
+
+const fetchLocations = async (params: $ReadOnlyArray<LocationInputID>) => {
+  const results = await fetch(
+    `/locations/id?limit=${params.length}&` +
+      params
+        .map((param: LocationInputID) => {
+          if (param.code !== undefined) {
+            const code = param.code ?? '';
+            return `id=${code}`;
+          }
+          return '';
+        })
+        .join('&'),
+  );
+  return results.locations.map(location => sanitizeLocations([location]));
+};
+
+export default function locationsByIDLoader() {
+  return new OptimisticDataloader(
+    (
+      ids: $ReadOnlyArray<LocationInputID>,
+    ): Promise<Array<Location[] | Error>> => fetchLocations(ids),
```

Comment:
Would it be possible that this returns `Promise<Array<Location | Error>>` instead. When querying for ID, you just expect to receive one result.